torch/_inductor/debug.py

```diff
@@ -387,3 +392,98 @@ def graph_diagram(self, nodes: SchedulerNodeList):
 
     def output_code(self, filename):
         shutil.copy(filename, self.filename("output_code.py"))
+
+
+@dataclasses.dataclass
+class TensorMetadataHolder:
+    tensor_metadata: TensorMetadata
+    device: torch.device
+
+
+save_args_cnt = itertools.count()
+
+
+def save_args_for_compile_fx_inner(fn):
+    """
+    This is a decorator applied on the compile_fx_inner function.
+    When config.save_args is True, this function will save the arguments
+    for each compile_fx_inner call to the filesystem. Later on
+    one can replay the compile_fx_inner call with the saved arguments
+    using load_args_and_run_compile_fx_inner.
+    """
+
+    assert (
+        fn.__name__ == "compile_fx_inner"
+    ), "This decorator only works for compile_fx_inner right now"
```

Comment:
Let's just put an extra `if ...:` inside `compile_fx_inner`, I think a decorator for just one function is harder to read (and results in ugly stack traces).